File: backend/StudentBlogAPI/Model/DTOs/UserDtos.cs
using System.ComponentModel.DataAnnotations;

namespace StudentBlogAPI.Model.DTOs;

public record UserResDto(
    int Id,
    string UserName,
    string FirstName,
    string LastName,
    string Email,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

public record UpdateUserInfoReqDto(
    string FirstName,
    string LastName
);

public record UpdatePasswordReqDto(
    [Required] string OldPassword,
    [Required] string NewPassword
);

File: frontend/types/converted-dtos/UserDtos.ts
export type UserResDto = {
  id: number;
  userName: string;
  firstName: string;
  lastName: string;
  email: string;
  createdAt: string; // Represents a DateTime type in string format
  updatedAt: string; // Represents a DateTime type in string format
};
export type UpdateUserInfoReqDto = {
  firstName: string;
  lastName: string;
};
export type UpdatePasswordReqDto = {
  oldPassword: string; // [Required]
  newPassword: string; // [Required]
};
